engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBean.java

```diff
@@ -6,32 +6,210 @@
+
+	public JavaBean(GetterPropertySelectionStrategy getterPropertySelectionStrategy, Class<?> clazz) {
+		this( getterPropertySelectionStrategy, clazz, false );
+	}
+
+	public JavaBean(GetterPropertySelectionStrategy getterPropertySelectionStrategy, Class<?> clazz, boolean includeNonDeclaredMethods) {
+		this.name = clazz.getName();
+		List<JavaBeanField> beanFields = new ArrayList<>();
+		for ( Field field : run( GetDeclaredFields.action( clazz ) ) ) {
+			if ( staticAndSyntheticFilter( field ) ) {
+				continue;
+			}
+			beanFields.add( new JavaBeanField( field ) );
+		}
+
+		List<JavaBeanConstructor> beanConstructors = new ArrayList<>();
+		for ( Constructor<?> constructor : run( GetDeclaredConstructors.action( clazz ) ) ) {
+			if ( staticAndSyntheticFilter( constructor ) ) {
+				continue;
+			}
+			beanConstructors.add( new JavaBeanConstructor( constructor ) );
+		}
+
+		List<JavaBeanGetter> beanGetters = new ArrayList<>();
+		List<JavaBeanMethod> beanMethods = new ArrayList<>();
+		Set<Method> processedMethods = new HashSet<>();
+		for ( Method method : run( GetDeclaredMethods.action( clazz ) ) ) {
+			processedMethods.add( method );
+			if ( staticAndSyntheticFilter( method ) ) {
+				continue;
+			}
+			processMethod( getterPropertySelectionStrategy, clazz, beanGetters, beanMethods, method );
+		}
+		if ( includeNonDeclaredMethods ) {
+			for ( Method method : run( GetMethods.action( clazz ) ) ) {
+				if ( processedMethods.contains( method ) || staticAndSyntheticFilter( method ) ) {
+					continue;
+				}
+				processMethod( getterPropertySelectionStrategy, clazz, beanGetters, beanMethods, method );
+			}
+		}
+
+		this.beanFields = CollectionHelper.toImmutableList( beanFields );
+		this.beanGetters = CollectionHelper.toImmutableList( beanGetters );
+		this.beanMethods = CollectionHelper.toImmutableList( beanMethods );
+		this.beanConstructors = CollectionHelper.toImmutableList( beanConstructors );
+	}
+
+	private static void processMethod(GetterPropertySelectionStrategy getterPropertySelectionStrategy, Class<?> clazz, List<JavaBeanGetter> beanGetters, List<JavaBeanMethod> beanMethods, Method method) {
+		JavaBeanConstrainableExecutable executable = new JavaBeanConstrainableExecutable( method );
+		if ( getterPropertySelectionStrategy.isProperty( executable ) ) {
+			beanGetters.add( new JavaBeanGetter( clazz, method, getterPropertySelectionStrategy.getPropertyName( executable ) ) );
+		}
+		else {
+			beanMethods.add( new JavaBeanMethod( method ) );
+		}
+	}
+
+	private boolean staticAndSyntheticFilter(Member member) {
+		// HV-172; ignoring synthetic members (inserted by the compiler), as they can't have any constraints
+		// anyway and possibly hide the actual member with the same signature in the built meta model
+		return Modifier.isStatic( member.getModifiers() ) || member.isSynthetic();
+	}
+
+	public Optional<JavaBeanMethod> getCallableByNameAndParameters(String methodName, Class<?>... parameterTypes) {
```

Comment:
Let's simplify all this. Here it should be `getDeclaredMethod()` (`Getter` is a specific case of `Method`).